### PR TITLE
Added getAvailableLanguagesFromResource() helper to i18n helpers

### PR DESCRIPTION
**Pull request by stephenwf**

### CODE REVIEW

**Comment by mathewjordan** [comment]
I have a few use cases for this @stephenwf  and I'd be happy to review.

**Comment by stephenwf** [comment]
Yeah that would be great, thanks @mathewjordan!

File: src/i18n.ts [hunk 1]
```diff
@@ -139,3 +140,64 @@ export function getValue(
     options
   );
 }
+
+function getLanguagesFromLanguageMap(languageMap: InternationalString) {
+  if (!languageMap) return [];
+  if (typeof languageMap === 'string') return [];
+  if (Array.isArray(languageMap)) return [];
+  return Object.keys(languageMap).filter((l) => l !== 'none');
```
Comment (mathewjordan) [inline src/i18n.ts:148]: 👍

**Review by mathewjordan — APPROVED** [review]
Love the simplicity of this. I have been intending to write something similar for a bit now. Glad this is available and thanks!

**Comment by stephenwf** [comment]
@mathewjordan I've stared at it for a while, but I don't know if I've missed any property that could have a language property. I know annotation bodies _could_ have language/value, but that feels like something else (not for making translating the IIIF manifest itself easier).

**Comment by stephenwf** [comment]
I think I want to add one more feature here. From `Vault` if you import a Manifest or Collection, I want it to add a `meta` value for the available languages.

So calling:
```ts
vault.getResourceMeta(manifest, 'availableLanguages'); // ['en', 'fr']
```

Maybe even with a helper on vault to do:
```ts
vault.getAvailableLanguages(manifest); // ['en', 'fr'];
```

**Comment by mathewjordan** [comment]
> I think I want to add one more feature here. From `Vault` if you import a Manifest or Collection, I want it to add a `meta` value for the available languages.
> 
> So calling:
> 
> ```ts
> vault.getResourceMeta(manifest, 'availableLanguages'); // ['en', 'fr']
> ```
> 
> Maybe even with a helper on vault to do:
> 
> ```ts
> vault.getAvailableLanguages(manifest); // ['en', 'fr'];
> ```

That looks very helpful.

**Comment by mathewjordan** [comment]
> @mathewjordan I've stared at it for a while, but I don't know if I've missed any property that could have a language property. I know annotation bodies _could_ have language/value, but that feels like something else (not for making translating the IIIF manifest itself easier).

I could see it useful for an Annotation body as well. Though, it'd be highly dependent on the motivation of the annotation and situation within the consuming app. In my use case, I would want to know all languages available of resources on screen, to provide a locale [Select] component.

**Comment by stephenwf** [comment]
Yeah, I think adding in a check on the Annotation body should be fairly straight-forward. I'll add that in.

**Pull request merged**